api/indexer.oas2.json

```
@@ -484,6 +484,98 @@
         }
       }
     },
+    "/v2/applications/{application-id}/boxes": {
+      "get": {
+        "description": "Given an application ID, returns the box names of that application sorted lexicographically.",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "tags": [
+          "search"
+        ],
+        "operationId": "searchForApplicationBoxes",
+        "summary": "Get box names for a given application.",
+        "parameters": [
+          {
+            "type": "integer",
+            "name": "application-id",
+            "in": "path",
+            "required": true
+          },
+          {
+            "$ref": "#/parameters/limit"
+          },
+          {
+            "type": "string",
+            "description": "The next page of results with strict lower bound given by box name in goal-arg form 'encoding:value'. Use the next token provided by the previous results.",
```

@tzaffi Minor:  Let me know if the proposal seems reasonable?

To be consistent with algod API docs, I propose replacing _goal-arg_ with _goal app call arg_.
* Source:  https://github.com/algorand/go-algorand/blob/feature/avm-box/daemon/algod/api/algod.oas2.json
* If a change is made here, there's several additional references in the PR that ought to be changed in tandem.
* If you feel _goal-arg_ is a better fit, then let's briefly discuss.  I want to land on a terminology we apply in indexer + algod consistently.

```suggestion
            "description": "The next page of results with strict lower bound given by box name in goal app call arg form 'encoding:value'. Use the next token provided by the previous results.",
```